### PR TITLE
Fix jackson deps and incrementalise

### PR DESCRIPTION
**Pull request by timja**
Without this change including the latest version of BFA as a plugin dep requires dependency management to be added to manage conflicts from `mongojack`, one of the pipeline plugins and `jackson2-api` plugin

Previously:
```
	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>com.fasterxml.jackson.core</groupId>
				<artifactId>jackson-annotations</artifactId>
				<version>${jackson.version}</version>
			</dependency>
			<dependency>
				<groupId>com.fasterxml.jackson.core</groupId>
				<artifactId>jackson-databind</artifactId>
				<version>${jackson.version}</version>
			</dependency>
		</dependencies>
	</dependencyManagement>

	<dependencies>
		<dependency>
			<groupId>com.sonyericsson.jenkins.plugins.bfa</groupId>
			<artifactId>build-failure-analyzer</artifactId>
			<version>1.24.0</version>
			<optional>true</optional>
		</dependency>
	</dependencies>
```

Now:
```
	<dependencies>
		<dependency>
			<groupId>com.sonyericsson.jenkins.plugins.bfa</groupId>
			<artifactId>build-failure-analyzer</artifactId>
			<version>1.24.1-SNAPSHOT</version>
			<optional>true</optional>
		</dependency>
	</dependencies>
```

Tested on the `claim` plugin

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -313,8 +317,8 @@
                     </dependency>
                 </dependencies>
                 <configuration>
-                    <configLocation>swe_checkstyle.xml</configLocation>
-                    <suppressionsLocation>checkstyle-suppressions.xml</suppressionsLocation>
+                    <configLocation>${project.basedir}/swe_checkstyle.xml</configLocation>
```
Comment (timja) [inline pom.xml]: required as part of incrementalising (so I can consume this PR in claim plugin if this takes time to merge)

**Comment by timja** [comment]
cc @rsandell @TWestling

**Comment by timja** [comment]
thanks,
any chance of a release please @rsandell?

**Comment by rsandell** [comment]
@timja I just released 1.24 and this PR is the only thing merged after that, so since it's only internal changes I don't see any reason to make a release. Incrementals should still work afaiu?

**Comment by timja** [comment]
> @timja I just released 1.24 and this PR is the only thing merged after that, so since it's only internal changes I don't see any reason to make a release. Incrementals should still work afaiu?

incrementals work but for the plugin to merge the changes to fix compatibility they want to depend on a released version, see https://github.com/jenkinsci/claim-plugin/pull/62

**Comment by rsandell** [comment]
https://github.com/jenkinsci/build-failure-analyzer-plugin/releases/tag/build-failure-analyzer-1.24.1

**Comment by timja** [comment]
Thanks, I've updated the PR

**Pull request merged**